### PR TITLE
Handling longitude base changes and masking appropriately

### PR DESCRIPTION
**Pull request by geflaspohler**
This PR implements a switch from base360 to base180 longitude representation, and allows easier slicing of dataframes, ploting of dataframes, and handling wrapping around the meridian in each representation.

### CODE REVIEW

File: sheerwater_benchmarking/baselines/__init__.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
 """Baseline models for the Sheerwater benchmarking project."""
-from .climatology import climatology
-from .ecmwf import iri_ecmwf, ecmwf_agg, ecmwf_rolled
+from .climatology import climatology, climatology_standard_30yr
+from .ecmwf import ecmwf_agg
 
 # Use __all__ to define what is part of the public API.
 __all__ = [
     climatology,
-    iri_ecmwf,
```
Comment (geflaspohler) [inline sheerwater_benchmarking/baselines/__init__.py:8]: I've made a bunch of these intermediate functions non-public. They're not all in base180 and really don't represent polished products. What do you think, @adkinsjd ?

File: sheerwater_benchmarking/baselines/climatology.py [hunk 2]
```diff
@@ -22,3 +22,14 @@ def climatology(first_year, last_year, variable, grid="global1_5", mask="lsm"):
...
+
+@dask_remote
+@cacheable(data_type='array',
+           cache_args=['variable', 'grid', 'mask'],
+           chunking={"lat": 121, "lon": 240, "doy": 366},
+           auto_rechunk=True)
+def climatology_standard_30yr(variable, grid="global1_5", mask="lsm"):
+    """Compute the standard 30-year climatology of ERA5 data from 1991-2020."""
+    # Get single day, masked data between start and end years
+    return climatology(1991, 2020, variable, grid=grid, mask=mask)
```
Comment (geflaspohler) [inline sheerwater_benchmarking/baselines/climatology.py:35]: Added a hard coded, simple standard climatology

File: sheerwater_benchmarking/baselines/ecmwf.py [hunk 3]
```diff
@@ -413,17 +421,23 @@ def ecmwf_agg(start_time, end_time, variable, forecast_type,
             - None: no mask
         verbose (bool): Whether to print verbose output.
     """
+    lons, lats, _ = get_grid(grid)
+    global_grid = get_global_grid(grid)
```
Comment (geflaspohler) [inline sheerwater_benchmarking/baselines/ecmwf.py:425]: I want to make ECMWF_rolled / ERA5_rolled only global functions, and the regional grids get applied at the final step. I'd love some feedback on the API here. I could also restrict the potential grids input into this function (this just gets the corresponding global grid, not matter what you pass).

File: sheerwater_benchmarking/baselines/ecmwf.py [hunk 4]
```diff
@@ -413,17 +421,23 @@ def ecmwf_agg(start_time, end_time, variable, forecast_type,
...
     ds = ecmwf_rolled(start_time, end_time, variable,
-                      forecast_type, grid=grid, agg=agg,
+                      forecast_type, grid=global_grid, agg=agg,
                       verbose=verbose)
 
+    # Convert to base180 longitude
+    ds = lon_base_change(ds, to_base="base180")
```
Comment (geflaspohler) [inline sheerwater_benchmarking/baselines/ecmwf.py:431]: Convert to base180 at this step (for the final ecmwf_agg output)

File: sheerwater_benchmarking/baselines/ecmwf.py [hunk 5]
```diff
@@ -413,17 +421,23 @@ def ecmwf_agg(start_time, end_time, variable, forecast_type,
...
         # Select variables and apply mask
-        mask_ds = land_sea_mask(grid=grid).compute()
+        mask_ds = land_sea_mask(grid=grid, base="base180").compute()
     elif mask is None:
         mask_ds = None
     else:
         raise NotImplementedError("Only land-sea or None mask is implemented.")
 
     ds = apply_mask(ds, mask_ds, variable)
+    ds = get_globe_slice(ds, lons, lats)
```
Comment (geflaspohler) [inline sheerwater_benchmarking/baselines/ecmwf.py:442]: Allows us to select a specific region of interest

File: sheerwater_benchmarking/masks.py [hunk 6]
```diff
@@ -3,21 +3,24 @@
...
-from sheerwater_benchmarking.utils import cacheable, cdsapi_secret, get_grid
+from sheerwater_benchmarking.utils import (cacheable, cdsapi_secret, get_grid, get_global_grid,
+                                           lon_base_change, get_globe_slice)
 
 
 @cacheable(data_type='array', cache_args=['grid'])
-def land_sea_mask(grid="global1_5"):
-    """Get the ECMWF land sea mask for the given grid.
+def land_sea_mask_global(grid="global1_5"):
+    """Get the ECMWF global land sea mask for the given grid.
```
Comment (geflaspohler) [inline sheerwater_benchmarking/masks.py:12]: Separated the global land sea mask from regional masks.

File: sheerwater_benchmarking/masks.py [hunk 7]
```diff
@@ -41,14 +44,45 @@ def land_sea_mask(grid="global1_5"):
 
     # Some transformations
     ds = xr.open_dataset(path)
+
     ds = ds.rename({'latitude': 'lat', 'longitude': 'lon', 'lsm': 'mask'})
-    ds = ds.sel(time=ds['time'].values[0])
-    ds = ds.drop('time')
+
+    if 'valid_time' in ds.coords:
```
Comment (geflaspohler) [inline sheerwater_benchmarking/masks.py:50]: Cleaning of the mask from the new CDS API, and changing to base180

File: sheerwater_benchmarking/masks.py [hunk 8]
```diff
@@ -41,14 +44,45 @@ def land_sea_mask(grid="global1_5"):
...
 
 
+@cacheable(data_type='array', cache_args=['grid'])
+def land_sea_mask(grid="global1_5"):
+    """Get the ECMWF land sea mask for the given grid.
+
+    Args:
+        grid (str): The grid to fetch the data at.
+    """
+    global_grid = get_global_grid(grid)
```
Comment (geflaspohler) [inline sheerwater_benchmarking/masks.py:80]: A regional, slicing land sea mask

File: pyproject.toml [hunk 9]
```diff
@@ -77,14 +77,17 @@ dependencies = [
...
     "dateparser",
     "bokeh",
     "flox",
     "ruff",
     "pytest",
-    "salientsdk",
+    "salientsdk>=0.3.4",
     "scikit-learn",
+    "xarray-regrid",
+    "weatherbench2 @ git+https://github.com/geflaspohler/weatherbench2",
```
Comment (adkinsjd) [inline pyproject.toml:90]: Redirect to rhiza github fork

**Pull request merged**